### PR TITLE
Support aws assume role

### PR DESCRIPTION
**Pull request by outtoin**
It support Amazon STS assume role (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html)

This PR require [code changes of aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp/pull/1590) to be merged.
After this merged, I'll replace GIT_REPOSITORY in build/CMakeLists.txt. (Currently, it use my repository.)

### CODE REVIEW

File: build/CMakeLists.txt [hunk 1]
```diff
@@ -503,13 +503,13 @@ ExternalProject_Add(cnmem
 #
 ExternalProject_Add(aws-sdk-cpp
   PREFIX aws-sdk-cpp
-  GIT_REPOSITORY "https://github.com/aws/aws-sdk-cpp.git"
-  GIT_TAG "1.7.129"
+  GIT_REPOSITORY "https://github.com/outtoin/aws-sdk-cpp.git"
+  GIT_TAG "origin/feature/log-stsprofile"
```
Comment (CoderHam) [inline build/CMakeLists.txt:507]: We usually want to stick to a tagged release and not a specific feature branch. If there is no release yet that contains this feature will the next release support it?

File: src/core/filesystem.cc [hunk 2]
```diff
@@ -1283,22 +1349,27 @@ S3FileSystem::S3FileSystem(
...
     config = Aws::Client::ClientConfiguration(profile_name);
   } else {
     config = Aws::Client::ClientConfiguration("default");
   }
 
+  if (region != NULL) {
+    config.region = region;
+  }
+
+  config.connectTimeoutMs = 30000;
```
Comment (CoderHam) [inline src/core/filesystem.cc:1370]: Why are you hardcoding these values and is thus necessary?

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: build/CMakeLists.txt (hunk 1 above)
Comment (outtoin) [reply build/CMakeLists.txt:507]: I submit PR to aws-sdk-cpp repository, but they doesn't review it... :(
After they merge my PR, new release contains this feature, I'll fix it.

File: src/core/filesystem.cc (hunk 2 above)
Comment (outtoin) [reply src/core/filesystem.cc:1370]: Nope. I will fix it.

**Comment by CoderHam** [comment]
@outtoin closing this due to inactivity. Please re-open if you are able to make the necessary changes.

**Pull request closed without merging**